android/src/main/kotlin/com/appcues/flutter/FlutterElementTargetingStrategy.kt

```diff
@@ -0,0 +1,158 @@
+        WindowInspector.getGlobalWindowViews().last()
+    } else {
+        @Suppress("SwallowedException", "TooGenericExceptionCaught")
+        try {
+            // this is the less desirable method for API 21-28, using reflection to try to get the root views
+            val windowManagerClass = Class.forName("android.view.WindowManagerGlobal")
+            val windowManager = windowManagerClass.getMethod("getInstance").invoke(null)
+            val getViewRootNames: Method = windowManagerClass.getMethod("getViewRootNames")
+            val getRootView: Method = windowManagerClass.getMethod("getRootView", String::class.java)
+            val rootViewNames = getViewRootNames.invoke(windowManager) as Array<Any?>
+            val rootViews = rootViewNames.map { getRootView(windowManager, it) as View }
+            rootViews.last()
+        } catch (_: Exception) {
+            // if all else fails, use the decorView on the window, which is typically the only one
+            window.decorView
+        }
+    }
+
+    return decorView.rootView as ViewGroup
+}
+
+internal class FlutterElementTargetingStrategy(val plugin: AppcuesFlutterPlugin) : ElementTargetingStrategy {
+
+    private var widgets: MutableMap<String, ViewElement> = mutableMapOf()
+
+    override fun captureLayout(): ViewElement? {
+        return plugin.activity?.getParentView()?.let {
+            val actualPosition = Rect()
+            it.getGlobalVisibleRect(actualPosition)
+
+            val displayMetrics = it.context.resources.displayMetrics
+            val density = displayMetrics.density
+
+            return ViewElement(
+                x = actualPosition.left.toDp(density),
+                y = actualPosition.top.toDp(density),
+                width = actualPosition.width().toDp(density),
+                height = actualPosition.height().toDp(density),
+                selector = null,
+                displayName = null,
+                type = this.javaClass.name,
+                children = widgets.values.toList(),
+            )
+        }
+    }
+
+    override fun inflateSelectorFrom(properties: Map<String, String>): ElementSelector? {
+        return FlutterElementSelector(
+            identifier = properties["identifier"],
+        ).let { if (it.isValid) it else null }
+    }
+
+    override fun captureScreenshot(): Screenshot? {
+        return plugin.activity?.let { activity ->
+            val view = activity.getParentView()
+            val density = view.resources.displayMetrics.density
+
+            // get the root level window insets, which will be used to apply a height correction
+            // to the screenshot below
+            val insets = ViewCompat.getRootWindowInsets(view)?.getInsets(WindowInsetsCompat.Type.systemBars())
+                ?: Insets.NONE
+
+            // This bitmap from the FlutterRenderer contains the top system bar but not the bottom
+            val bitmap = plugin.renderer.bitmap
```

Comment:
this is the custom screenshot part on Android - using the FlutterRenderer `bitmap` to get access - and some related special logic around sizing